sql/catalyst/src/main/java/org/apache/spark/sql/types/PhysicalDataTypes.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.types;
+
+import org.apache.spark.sql.catalyst.types.*;
+
+/**
+ * To get specific physical data type.
+ *
+ * @since 3.4.0
+ */
+public class PhysicalDataTypes {
```

Comment:
is this really needed? `PhysicalDataType` is an internal API, and we don't need to make it java friendly. We can write `df isinstanceof PhysicalStringType`.

Reply:
For example, in the following scenario:

https://github.com/apache/spark/blob/3fc8a90267312fdfc42b7e6a16ee69f2507ef56b/sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java#L79-L95

if we can replace `PhysicalDataType` with `DataType`:

```java
 mutableFieldTypes = Collections.unmodifiableSet(
      new HashSet<>(
        Arrays.asList(
          PhysicalNullType$.MODULE$,
          PhysicalBooleanType$.MODULE$,
          PhysicalByteType$.MODULE$,
          PhysicalShortType$.MODULE$,
          PhysicalIntegerType$.MODULE$,
          PhysicalLongType$.MODULE$,
          PhysicalFloatType$.MODULE$,
          PhysicalDoubleType$.MODULE$
        )));
```

Does it look ugly

Reply:
If it's not ugly, I think `PhysicalDataTypes` can be deleted

Reply:
It's OK. This is internal code anyway.

Reply:
OK, let me remove PhysicalDataTypes and change to use `isinstanceof `

Reply:
done